### ISSUE TITLE
Training very small custom data in yolov5, load pre-trained weight and freeze some layers?

### ISSUE THREAD

**Issue by JunFang-NWPU** [ultralytics/yolov5#49]
Hi,

If I have only a very small dataset, e.g., server hundred images for about ten classes. As suggested, i) I need to train my own data from the pre-trained weight, and ii) maybe I need to freeze some layers, for instance , the backbone, and only modify weight of other layers.

My questions are: 
i) Are the above two methods are the best strategies to deal with training very small dataset?
ii) How to freeze some layers of the yolov5?

Thanks.

**Comment by glenn-jocher** [comment]
@JunFang-NWPU for training very small datasets, my number one recommendation is to not train, and instead collect more training data. The chances of your results generalizing well are not good.

If you want to continue training with a small dataset, then simply start from pretrained weights and train normally. --adam sometimes helps smaller datasets as well, though it generally benefits from a smaller learning rate than SGD, so if you use it I would reduce your `hyps['lr0']` by 10 (though I have not tested it yet in this repo, so anything might happen).
```
python train.py --cfg yolov5s.yaml --weights yolov5s.pt
```

I always advise against freezing any layers. I've made this point many times before in ultralytics/yolov3, and the same applies here. If you still want to freeze layers, you can simply set the `requires_grad` of any parameters you want to freeze to `False`.

**Comment by JunFang-NWPU** [comment]
OK, Thanks. I will give a try.
